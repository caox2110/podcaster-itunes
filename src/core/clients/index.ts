export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
