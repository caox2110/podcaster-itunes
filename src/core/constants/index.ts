import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
