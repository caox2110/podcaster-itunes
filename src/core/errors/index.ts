export { default as ExceptionHandler } from './Exceptions.handler';
