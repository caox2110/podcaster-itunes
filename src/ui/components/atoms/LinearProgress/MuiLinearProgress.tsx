import LinearProgress from '@mui/material/LinearProgress';
import type { LinearProgressProps } from '@mui/material/LinearProgress';

/**
 * LinearProgress Component definition.
 *
 * @export
 * @param {LinearProgressProps} { ...rest }
 * @returns {*}
 */
export default function MuiLinearProgress({ ...rest }: LinearProgressProps) {
  return <LinearProgress {...rest} />;
}
