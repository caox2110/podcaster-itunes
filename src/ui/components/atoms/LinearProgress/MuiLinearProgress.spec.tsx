import type { LinearProgressProps } from '@mui/material/LinearProgress';

import { render } from '@/core/test';
import type { RenderResult } from '@/core/test';

import MuiLinearProgress from './MuiLinearProgress';

const properties = {};

function renderApp({ ...rest }: LinearProgressProps = properties): RenderResult {
  const utils = render(<MuiLinearProgress {...rest} />, { withRouter: true });

  return {
    ...utils,
  };
}

describe('MuiLinearProgress', () => {
  describe('Rendering', () => {
    it('Should render without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
