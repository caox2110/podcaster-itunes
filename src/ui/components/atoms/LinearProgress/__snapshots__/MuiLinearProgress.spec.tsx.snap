// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiLinearProgress Rendering Should render without throwing any errors 1`] = `
<div>
  <span
    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
    />
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
    />
  </span>
</div>
`;
